frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.tsx

```diff
@@ -260,18 +263,20 @@ const TableRow = ({
       <a
         className={cx(
           "AdminList-item flex align-center no-decoration text-wrap justify-between",
-          { selected: isSelected },
+          { selected: isSelected, disabled: !isSyncCompleted(table) },
```

Comment:
This uses a very old CSS declaration

https://github.com/metabase/metabase/blob/86c4848035b23027ff11ef84b6fec1a29a7f6c53/frontend/src/metabase/css/admin.css#L119-L122

could you maybe migrate this to styled component and pass `disabled` prop directly? The pros are that it will also automatically get `disabled` prop on the dom node, and you could assert in Cypress using `should("be.disabled")` instead.

Just to be clear you can use this style in the styled component file (there's none, so you need to create a new file)

```ts
   color: ${color("text-medium")}; 
   pointer-events: none; 
   opacity: 0.4;
```

These are the styles I got from inspecting the element. You should double check.

Reply:
Okay, I think tried to create a new styled component in [Create AdminListItem styled component](https://github.com/metabase/metabase/pull/31661/commits/e9708e0ea589ad165d91c1197a848745a53038c3). 

One issue is I don't know how to prevent duplication when converting the `selected` class into a prop.

This was in the css before:
```
.AdminList-item.selected,
.AdminList-item:hover {
  background-color: white;
  border-color: var(--color-border);
  margin-left: -0.5em;
  margin-right: -0.5em;
  padding-left: 1.5em;
  padding-right: 1.5em;
  box-shadow: 0 1px 2px var(--color-shadow);
}
```

And now I have duplicated the css in the `AdminListItem` styled component:
```
  ${({ selected }) =>
    selected &&
    css`
      background-color: white;
      border-color: var(--color-border);
      margin-left: -0.5em;
      margin-right: -0.5em;
      padding-left: 1.5em;
      padding-right: 1.5em;
      box-shadow: 0 1px 2px var(--color-shadow);
    `};
  &:hover {
    background-color: white;
    border-color: color("border");
    margin-left: -0.5em;
    margin-right: -0.5em;
    padding-left: 1.5em;
    padding-right: 1.5em;
    box-shadow: 0 1px 2px ${color("shadow")};
  }
```

I could remove the duplication by extracting the css props out to a varaible and name it something like `selectedOrHoverStyle`, but I'd like to avoid having to name it and extract it out. Is there another way?

Reply:
At first, I was thinking about only migrating `disabled`–which is what you've added–but the entire component is even better.

Unfortunately there's no way to go around this as far as I can tell, previously we can use the same block since we use 2 CSS selectors, but since we're passing props directly we can't really combine dynamic style + CSS selector, in this case, `selected` + `:hover`.